luigi/worker.py

```diff
@@ -821,11 +822,27 @@ def _keep_alive(self, n_pending_tasks, n_unique_pending):
         """
         if not self._config.keep_alive:
             return False
+        if not self._is_alive():
+            return False
         elif self._assistant:
             return True
         else:
             return n_pending_tasks and (n_unique_pending or not self._config.count_uniques)
 
+    def _is_alive(self):
+        """
+        Returns True if this worker is alive.
+
+        Returns False if the keep-alive thread has been stopped.
+        Returns False if the worker has not been initialized using a with statement.
+        Otherwise, this will return true.
+        """
+        try:
+            return self._keep_alive_thread.isAlive()
```

Comment:
Shouldn't you prefer the [`is_alive()` spelling](https://docs.python.org/2/library/threading.html#threading.Thread.is_alive)?